### ISSUE TITLE
Clan/Group System

### ISSUE THREAD

**Issue by RileyNiko** [o5k/tetrio-issues#1361]
### Checklist

- [X] I've **checked both [open and recently closed issues](https://github.com/tetrio/issues/issues?q=is%3Aissue)** to ensure this feature is not already requested.
- [X] I am using the **newest version** of TETR.IO.
- [X] I am certain this is a **feature request**, not a bug.
- [X] I've **thought this feature through** properly.

### Is your feature request related to a problem? Please describe.

Playing with friends in TETR.IO is pretty fun, but competing with each other asynchronously is tough. You could post your scores in the group chat or just talk about them next time you're online, but that also means every time you score you need to check all your friends to see if you're at the top.

### Describe the solution you'd like

Some sort of clan or group system to compare scores or even compete against other groups.

TETR.IO already has a "Global" and a "Country" placement system. I propose that another leaderboard could be tracked of whatever group a player is a part of. This could then take advantage of the "Country Best" or "World Record" splash-text that pops up, instead saying "Group Best" or "Clan Record". 

Furthermore, this could lead to a sort of clan ranking system, although I'm unsure what metric could be unified to signify the "strength" of a clan on some sort of leaderboard. I mostly just made this for the intercompetition aspect, not intracompetition. 

### Describe alternatives you've considered

A simpler alternative is a basic leaderboard underneath the leaderboard option when selecting a mode.
![image](https://github.com/user-attachments/assets/104e354d-7566-486b-94e2-097b8f87e157)
It's not as robust as a solution, but it would solve the idea that sparked this entire request.

**Comment by o5k** [comment]
TETR.IO does have friend leaderboards! In TETRA CHANNEL (ingame), you can click into Leaderboards or Players and filter by your friends only. This lists everyone you have added to your friends list.

In addition, we plan to send out ingame notifications when a friend beats your score, to make it easier to stay in the loop.

**Comment by RileyNiko** [comment]
OH MY GOD SDLJFNBSLDFBLSKJDB IM SO DUMB BUT THANK YOU

**Comment by CreepercraftYT** [comment]
being able to make clans/groups would be a nice idea imo

**Comment by furuochen-dev** [comment]
I agree that this would be very important, as currently the game lacks a bit of social aspect.
